### ISSUE TITLE
False positive for the useless_attribute rule

### ISSUE THREAD

**Issue by Recmo** [rust-lang/rust-clippy#5918]
I tried this code:

```rust
#[allow(clippy::wildcard_imports)]
use std::prelude::v1::*;
```

I expected to see this happen: *explanation*

Nothing, since this is an exception to the wildcard_imports rule, required because of #5917.

Instead, this happened: *explanation*

```
error: useless lint attribute
 --> utils/mmap-vec/src/mmap_vec.rs:5:1
  |
5 | #[allow(clippy::wildcard_imports)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(clippy::wildcard_imports)]`
  |
  = note: `-D clippy::useless-attribute` implied by `-D warnings`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute
```

### Meta

- `cargo clippy -V`: clippy 0.0.212 (7e6d6e5f5 2020-08-16)
- `rustc -Vv`:
  ```
  rustc 1.47.0-nightly (7e6d6e5f5 2020-08-16)
  binary: rustc
  commit-hash: 7e6d6e5f535321c2223f044caba16f97b825009c
  commit-date: 2020-08-16
  host: x86_64-apple-darwin
  release: 1.47.0-nightly
  LLVM version: 10.0
  ```

<details><summary>Backtrace</summary>
  <p>
  
  ```
  <backtrace>
  ```
  
  </p>
</details>

**Comment by Recmo** [comment]
Same for `#[allow(clippy::enum_glob_use)]`
